website_no_crawler_environment/views/disable_robots.xml

```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
+    <template id="robots" inherit_id="website.robots">
+        <xpath expr="t" position="replace">
+            <t t-translation="off">
+                User-agent: *
+                <t t-if="server_env in ('preprod', 'stage')">Disallow: /</t>
```

Comment:
In this way you're limiting user to have `preprod` and `stage` env only. Better to generalize it, having `test` as default one.